department-of-veterans-affairs/abd-vro
# Upgrade Spring 6/Spring Boot 3 for non-camel containers

## What was the problem?

Out-of-date spring framework and spring boot versions were being used in containers. 

Associated tickets or Slack threads:

- #1428 

## How does this fix it?[^1]

For containers that do not make use of Camel, latest versions of the dependencies implementation are included in the build script to force upgrades to the latest Spring/Spring Boot versions.

Right now, the only containers that do not make use of Camel but do make use of Spring is `svc-lighthouse` and `svc-bip`.

I verified that latest versions of Spring dependencies are being pulled in by running the Gradle dependencies tasks for the two projects given above.

## How to test this PR
- Passing unit tests
- Successfully starting containers on local machine
- Successful deployment and healthy containers in `dev` LHDI environment
- Passing SecRel run on the two images to further confirm that outdated spring versions are not being used


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
